x-pack/plugins/ml/public/data_frame/components/define_pivot/pivot_preview.tsx

```diff
@@ -32,36 +43,58 @@ export const PivotPreview: React.SFC<Props> = React.memo(({ aggs, groupBy, query
     return null;
   }
 
-  const [loading, setLoading] = useState(false);
-  const [dataFramePreviewData, setDataFramePreviewData] = useState([]);
-
-  useEffect(
-    () => {
-      if (aggs.length === 0) {
-        setDataFramePreviewData([]);
-        return;
-      }
-
-      setLoading(true);
-
-      const request = getDataFramePreviewRequest(indexPattern.title, query, groupBy, aggs);
-
-      ml.dataFrame
-        .getDataFrameTransformsPreview(request)
-        .then((resp: any) => {
-          setDataFramePreviewData(resp.preview);
-          setLoading(false);
-        })
-        .catch((resp: any) => {
-          setDataFramePreviewData([]);
-          setLoading(false);
-        });
-    },
-    [indexPattern.title, aggs, groupBy, query]
+  const { dataFramePreviewData, errorMessage, status } = usePivotPreviewData(
+    indexPattern,
+    query,
+    aggs,
+    groupBy
   );
 
+  if (status === PIVOT_PREVIEW_STATUS.ERROR) {
+    return (
+      <EuiPanel grow={false}>
+        <PreviewTitle />
+        <EuiCallOut
+          title={i18n.translate(
+            'xpack.ml.dataframe.sourceIndexPreview.dataFramePivotPreviewError',
+            {
+              defaultMessage: 'An error occurred loading the pivot preview.',
+            }
+          )}
+          color="danger"
+          iconType="cross"
+        >
+          <p>{errorMessage}</p>
+        </EuiCallOut>
+      </EuiPanel>
+    );
+  }
+
   if (dataFramePreviewData.length === 0) {
-    return null;
+    return (
+      <EuiPanel grow={false}>
+        <PreviewTitle />
+        <EuiCallOut
+          title={i18n.translate(
+            'xpack.ml.dataframe.sourceIndexPreview.dataFramePivotPreviewNoDataCalloutTitle',
+            {
+              defaultMessage: 'Pivot preview not available',
+            }
+          )}
+          color="primary"
+        >
+          <p>
+            {i18n.translate(
+              'xpack.ml.dataframe.sourceIndexPreview.dataFramePivotPreviewNoDataCalloutBody',
+              {
+                defaultMessage:
+                  'Please make sure to choose at least one group-by field and aggregation.',
```

Comment:
I'd change this to just `Please choose at least one group-by field and aggregation.`, particularly as this is shown on opening the page before the user has had a chance to select the fields.

Reply:
Changed in a5ee6fb.